src/modules/workspaces/interceptors/workspaces-logs.interceptor.ts

```diff
@@ -0,0 +1,332 @@
+    const request = context.switchToHttp().getRequest();
+    this.logAction = this.getWorkspaceLogAction(context);
+
+    if (!Object.values(WorkspaceLogType).includes(this.logAction)) {
+      Logger.debug(`[WORKSPACE/LOGS] Invalid log action: ${this.logAction}`);
+      return;
+    }
+
+    const platform = this.determinePlatform(request.headers['internxt-client']);
+
+    return next.handle().pipe(
+      mergeMap(async (res) => {
+        await this.handleAction(platform, this.logAction, request, res);
+      }),
+    );
+  }
+
+  async logIn(platform: WorkspaceLogPlatform, req: any, res: any) {
+    await this.handleUserAction(platform, WorkspaceLogType.LOGIN, req, res);
+  }
+
+  async changedPassword(platform: WorkspaceLogPlatform, req: any, res: any) {
+    await this.handleUserAction(
+      platform,
+      WorkspaceLogType.CHANGED_PASSWORD,
+      req,
+      res,
+    );
+  }
+
+  async logout(platform: WorkspaceLogPlatform, req: any, res: any) {
+    await this.handleUserAction(platform, WorkspaceLogType.LOGOUT, req, res);
+  }
+
+  async share(platform: WorkspaceLogPlatform, req: any, res: any) {
+    const itemType = this.getItemType(req);
+    if (!itemType) {
+      Logger.debug('[WORKSPACE/LOGS] The item type is required');
+      return;
+    }
+
+    const action = this.determineAction(
+      'SHARE',
+      itemType as unknown as WorkspaceItemType,
+    );
+    if (action) {
+      await this.handleUserWorkspaceAction(platform, action, req, res);
+    }
+  }
+
+  async shareFile(platform: WorkspaceLogPlatform, req: any, res: any) {
+    await this.handleUserWorkspaceAction(
+      platform,
+      WorkspaceLogType.SHARE_FILE,
+      req,
+      res,
+    );
+  }
+
+  async shareFolder(platform: WorkspaceLogPlatform, req: any, res: any) {
+    await this.handleUserWorkspaceAction(
+      platform,
+      WorkspaceLogType.SHARE_FOLDER,
+      req,
+      res,
+    );
+  }
+
+  async delete(platform: WorkspaceLogPlatform, req: any, res: any) {
+    const items: DeleteItem[] = req?.body?.items || req?.items;
+    if (!items || items.length === 0) {
+      Logger.debug('[WORKSPACE/LOGS] The items are required');
+      return;
+    }
+
+    const { ok, requesterUuid, workspaceId } = this.extractRequestData(req);
+
+    if (ok) {
+      const deletePromises = items
+        .map((item) => {
+          const action = this.determineAction(
+            'DELETE',
+            item.type as unknown as WorkspaceItemType,
+          );
+          if (action) {
+            return this.registerWorkspaceLog({
+              workspaceId: workspaceId,
+              creator: requesterUuid,
+              type: action,
+              platform,
+              entityId: item.uuid || item.id,
+            });
+          }
+          return null;
+        })
+        .filter((promise): promise is Promise<any> => promise !== null);
+      await Promise.all(deletePromises);
```

Comment:
This function also applies when I empty my trash, right?  (I am just guessing as there is a DELETE_ALL action)

What happens if user has 1000 items inside the trash? Are we going to send 1000 inserts to the database at the same time? This is something we could discuss with business